### DISCUSSION TITLE
Fast refresh isn't working in custom express server with typescript.

### DISCUSSION THREAD

**Discussion by imranbarbhuiya** [vercel/next.js#30196] [Help] [answered]
I'm using nextjs with my custom express server in typescript. But I'm facing a bunch of problems.

1. After editing anything, it shows `compiled successfully`, but it's not refreshing the page. I need to manually refresh the page to see changes. 
2. It's adding a style 
```html
<style data-next-hide-fouc="true">body{display:none}</style>
```
Which isn't removing. 
3. CSS import aren't working, only inline/internal CSS are working + adding link tag inside Head component also working.

But everything works fine in production mode.

I've found that many other peoples also had this issue, but I'm not finding a proper solution.
[#10285](https://github.com/vercel/next.js/issues/10285)
One solution is by adding `body{display:block;}` in _document.tsx which only solves display none issue but no CSS modules are loading.

I've tried to remove all middlewares, removing all express routes but still it's not working.

**Comment by jamesmosier** [comment]
Hi @imranbarbhuiya, do you happen to have a repository that reproduces this issue?

**Reply by imranbarbhuiya** [reply]
Yeah, btw I was making it in express that's why you'll find ejs also I'm just converting it to nextjs. [Repo](https://github.com/imranbarbhuiya/Codversity)

**Reply by imranbarbhuiya** [reply]
Also, Just adding It might help. I'd added a wrong style tag inside the index page then browser was showing me error **(without `display:none`)** as well as fast refresh was also working but as soon as I fixed that tag again the page was loaded with `display:none` and fast refresh wasn't working again.

**Comment by imranbarbhuiya** [comment]
@jamesmosier help please

**Comment by imranbarbhuiya** [comment, marked as answer]
After upgrading to v12, It's fixed now, thank you.
